src/postgres/postgresConnectionStrings.ts

```diff
@@ -27,13 +27,14 @@ export class ParsedPostgresConnectionString extends ParsedConnectionString {
     public username: string | undefined;
     public password: string | undefined;
     public readonly port: string;
+    public readonly options: string;
 
     constructor(connectionString: string, config: ConnectionOptions) {
         super(connectionString, config.database ? config.database : undefined);
         this.hostName = nonNullProp(config, 'host');
         this.port = config.port ? config.port : `${postgresDefaultPort}`;
-        this.username = config.user;
-        this.password = config.password;
+        this.username = nonNullProp(config, 'user');
```

Comment:
These shouldn't be using `nonNullProp` since user/password are optional. I looked into it and seems like we're just getting lucky that `parse` is setting it to an empty string instead of `undefined`